File: leetcode/045-jump-game-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
  var jumps = 0;
  var a = 0, b = 0;
  for (var i = 0; i < nums.length; ++i) {
    b = Math.max(nums[i] + i, b);
    if (a + 1 >= nums.length) break;
    if (i == a) a = b, ++jumps;
  }
  return jumps;
};

File: leetcode/045-jump-game-ii.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        jumps, a, b, n = 0, 0, 0, len(nums)
        for i, num in enumerate(nums):
            b = max(num + i, b)
            if a + 1 >= n: break
            if i == a:
                a = b
                jumps += 1
        return jumps
